### PR TITLE
docs: Update requirements & tests manual

### PR DESCRIPTION
**Pull request by scyzoryck**
Hello!
My first PR here. I made some small improvements to the setup documentation. 
However, I might need some help with `Running the tests` section - After running `tests/script_jackrabbit.sh` it fails on the step of missing composer dependencies. 

Best, scyzoryck.

### CODE REVIEW

File: tests/script_jackrabbit.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ elif test "${SCRIPT_DIR:0:1}" != "/" ; then
   SCRIPT_DIR="$PWD/$SCRIPT_DIR"
 fi
 
-./tests/jackrabbit.sh
+./tests/jackrabbit.sh || exit
```
Comment (dbu) [inline tests/script_jackrabbit.sh:16]: why this change? should we do `set -e` at the top of the script to abort on any error?

File: README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 ## Requirements
 
-* PHP 7.1
+* PHP 7.4
```
Comment (dbu) [inline README.md]:
```suggestion
```

that is defined in composer.json which we mention below, imho thats enough.

**Review by dbu — COMMENTED** [review]
thanks a lot for the cleanup! seems we forgot about the documentation when we moved away from travis-ci.

**Comment by scyzoryck** [comment]
Thanks @dbu for suggestions! I rewrote a bit section about setting up package for tests based on the commands from Github actions. :) Do you have some time for another look, please?

File: README.md [hunk 3]
```diff
@@ -29,22 +28,38 @@ Thanks to
...
-    ./tests/travis_jackrabbit.sh
-
-Please note that this will also require implementations for virtual packages
-that match the backend you want to test, resulting in a change in your
-`composer.json`. Make sure you do not check in this change into version control.
-
-3. Run the tests:
-
-    phpunit -c tests/phpunit_jackrabbit.xml.dist
+There are separate test sets for `dbal` and `jackrabbit` backends.
```
Comment (dbu) [inline README.md]:
```suggestion
There are separate test sets for the `doctrine-dbal` and the `jackrabbit` backend.
```

**Comment by dbu** [comment]
i realized that the documentation will be easier to read when explaining by implementation each step, rather than each step for each implementation. i reformatted it in #842 

thanks to your work, this was quite easy to do now as you collected the necessary information and steps. thanks a lot!
